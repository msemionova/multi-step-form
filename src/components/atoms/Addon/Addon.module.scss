.addon {
  display: flex;
  align-items: center;
  gap: 2.4rem;
  background-color: var(--color-white);
  border-radius: var(--border-radius-medium);
  border: var(--border-secondary);
  transition: background-color, border 200ms;
  padding: 2rem 1.6rem;
  margin-bottom: 1.4rem;
  cursor: pointer;

  h3 {
    font: var(--typography-input-text);
  }

  &::before {
    content: '';
    width: 2rem;
    height: 2rem;
    background-color: var(--color-white);
    border-radius: var(--border-radius-small);
    border: var(--border-secondary);
  }
}

input[type="checkbox"] {
  display: none;
}

input[type="checkbox"]:checked + .addon {
  background-color: var(--color-gray-background);
  border: var(--border-primary);

  &::before {
    background: no-repeat center/60% url("~/public/assets/images/icon-checkmark.svg");
    background-color: var(--color-primary-dark);
    border: var(--border-primary);
  }
}

.description,
.price {
  font: var(--typography-input-label);
}

.description {
  color: var(--color-gray-text-light);
}

.price {
  text-align: right;
  flex: 1 1 auto;
  color: var(--color-primary-dark);
}
