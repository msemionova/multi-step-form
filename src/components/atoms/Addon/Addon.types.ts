export type AddonProps = {
  id: string,
  title: string,
  description: string,
  priceMonthly: number,
  priceYearly: number,
  type?: 'monthly' | 'yearly',
  currency?: string,
  isChecked?: boolean
}
