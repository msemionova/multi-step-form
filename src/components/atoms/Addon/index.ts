export { default as Addon } from './Addon';
export type { AddonProps } from './Addon.types';
