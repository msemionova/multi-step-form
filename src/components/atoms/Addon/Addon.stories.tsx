import type { Meta, StoryObj } from '@storybook/react';

import Addon from './Addon';

const meta: Meta<typeof Addon> = {
  component: Addon,
};

export default meta;
type Story = StoryObj<typeof Addon>;

export const AddonChecked: Story = {
  args: {
    id: 'test_id',
    title: 'Addon One',
    description: 'Addon description',
    priceMonthly: 2,
    priceYearly: 20,
    isChecked: true
  }
};

export const AddonUnchecked: Story = {
  args: {
    id: 'test_id',
    title: 'Addon One',
    description: 'Addon description',
    priceMonthly: 2,
    priceYearly: 20
  }
};
