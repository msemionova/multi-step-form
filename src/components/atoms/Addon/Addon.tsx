import styles from './Addon.module.scss';
import { AddonProps } from './Addon.types';

const Addon = ({
  id,
  title,
  description,
  priceMonthly,
  priceYearly,
  type = 'monthly',
  currency = '$',
  isChecked = false
}: AddonProps) => {
  const price: string =
    type === 'monthly' ? `${String(priceMonthly)}/mo` : `${String(priceYearly)}/yr`;

  return (
    <div>
      <input type='checkbox' name='addons' id={id} value={id} defaultChecked={isChecked} />
      <label className={styles.addon} htmlFor={id}>
        <div>
          <h3>{title}</h3>
          <span className={styles.description}>{description}</span>
        </div>
        <span className={styles.price}>{`+${currency}${price}`}</span>
      </label>
    </div>
  );
};

export default Addon;
