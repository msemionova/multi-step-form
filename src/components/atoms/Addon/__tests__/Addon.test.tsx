import Addon from '../Addon';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

describe('<Addon />', () => {
  it('renders correctly without required nor label...', () => {
    const dummyProps = {
      id: 'addon',
      title: 'Addon Title',
      description: 'Addon Description',
      priceMonthly: 9,
      priceYearly: 999,
      isChecked: true
    };

    const tree = renderer
      .create(<Addon {...dummyProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
