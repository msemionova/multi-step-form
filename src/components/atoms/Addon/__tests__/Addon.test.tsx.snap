// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Addon /> renders correctly without required nor label... 1`] = `
<div>
  <input
    defaultChecked={true}
    id="addon"
    name="addons"
    type="checkbox"
    value="addon"
  />
  <label
    className="addon"
    htmlFor="addon"
  >
    <div>
      <h3>
        Addon Title
      </h3>
      <span
        className="description"
      >
        Addon Description
      </span>
    </div>
    <span
      className="price"
    >
      +$9/mo
    </span>
  </label>
</div>
`;
